docs_src/design/ucr/0005-Hybrid-App-Device-Services.md

```diff
@@ -0,0 +1,179 @@
+## Hybrid App-Device Services
+This UCR describes Use Cases for an approach that is useful for some analytic, utility, and north-bound services 
+which are formed from a hybrid of EdgeX App and Device Services.
+
+### Submitters
+- Tom Brennan (Eaton)
+
+## Change Log
+- [pending](https://github.com/edgexfoundry/edgex-docs/pulls) (2022-07-26)
+
+
+### Market Segments
+Any that deploy EdgeX systems with analytic, utility, or north-bound microservices that run persistently
+and rely on device profiles to describe their handling of data from either south-bound devices or 
+service-based devices.
+
+
+### Motivation
+The EdgeX docs describe a nice, clean framework wherein there are south-bound Device Services that communicate with
+external devices and import their data, and north-bound Application Services that transform that
+device data for external actors; there are separate SDKs for Device and Application Services, and
+there are separate sets of APIs associated with Device and Application Services. In this view, the work of 
+Application Services is triggered by internal EdgeX events (API calls or Message Bus events).
+
+In our implementations, we find several problems with clean separation of Device and Application Services as we 
+design microservices for our industrial products:
+- Our north-bound microservices respond to external, as well as internal, stimuli, and need to maintain
+persistent connections to external actors or to listen persistently for external requests.
+- For analytic or north-bound services managing large amounts of data from south-bound devices, the scale does 
+not permit the use of per-resource rules to describe how each resource is handled; instead, it is necessary to 
+code the rules and let the bulk or batch handling be described by something just like device profiles. 
+These are the 95-98% case, where the coded rules are relatively simple (eg, for interval trending or alarm 
+monitoring); for the remainder, the complex rule cases, the existing EdgeX means of rule handling (triggers and 
+eKuiper rules) are a good solution.
+- Static configuration of analytic and north-bound services with respect to the device data that they must handle
+has proven insufficient; instead, these services need something just like the dynamic provisioning of Device Services.
+- For our analytic, utility, and north-bound services, we almost always choose to add (Device) Resources as the 
+means to manage their configuration, transforms, and status reporting. These Resources are often needed on a 
+per-Device basis (rather than just overall service configuration or status), sometimes extending (adding to) the 
```

Comment:
Please clarify `per-Device instance basis` or `per-Device class basis`

Reply:
Please give example of `north-bound protocol 
libraries which lack a Go implementation`

Reply:
> Please clarify per-Device instance basis or per-Device class basis

Kind of both: per Device class when described via a Device Profile, but then instantiated into per Device instances, like other device services do.  I'll clarify and state "per-Device instance basis".

> Please give example of north-bound protocol libraries which lack a Go implementation

Good point; I shouldn't just whine. I know that for Azure, there is a Go implementation for IoT Hub connections but not for Azure Device Provisioning. 
While google points to examples of Golang libraries for BACnet, they are not well supported, up to date, or have been through the rigorous and expensive certification process (ie, I wouldn't bet a project's success on them), whereas there are known good C/C++ BACnet libraries. I suspect the same story for other protocols like Ethernet/IP and DNP3.